File: mobile-client/ios/Sources/CoreSecurity/Security.swift
import Foundation

public struct Security {
  private let applyRequestSignatureUseCase: ApplyRequestSignatureUseCase
  private let requestSignatureApplier: RequestSignatureApplier

  public init(secret: String = "secret") {
    let applier = RequestSignatureApplier(secret: secret)
    self.requestSignatureApplier = applier
    self.applyRequestSignatureUseCase = ApplyRequestSignatureUseCase(requestSignatureApplier: applier)
  }

  public func headers(for path: String) -> SignatureHeaders {
    requestSignatureApplier.headers(for: path)
  }

  public func applySignature(to req: URLRequest) -> URLRequest {
    applyRequestSignatureUseCase(request: req)
  }
}

File: mobile-client/android/core/security/src/main/java/com/smartresponsor/core/security/Security.kt
package com.smartresponsor.core.security

import com.smartresponsor.mobile.client.contract.system.security.SignatureHeaders
import com.smartresponsor.mobile.client.data.system.security.RequestSignatureApplier
import com.smartresponsor.mobile.client.usecase.system.security.ApplyRequestSignatureUseCase
import okhttp3.Request

/**
 * Legacy-compatible Android entry point bridged to canonical system/security slices.
 */
class Security(
    secret: String,
) {
    private val requestSignatureApplier: RequestSignatureApplier = RequestSignatureApplier(secret)
    private val applyRequestSignatureUseCase: ApplyRequestSignatureUseCase =
        ApplyRequestSignatureUseCase(requestSignatureApplier)

    fun sign(data: String): String = requestSignatureApplier.sign(data)

    fun headersFor(path: String): SignatureHeaders = requestSignatureApplier.headersFor(path)

    fun apply(request: Request): Request = applyRequestSignatureUseCase(request)
}
